src/main/java/model/CodeQualityRule.java

```diff
@@ -0,0 +1,52 @@
+package main.java.model;
+
+/**
+ * A class that represents a rule that can be applied to
+ * the excel metrics.
+ * 
+ * @author Lino Silva
+ */
+public class CodeQualityRule {
+
+	private String name;
+	private boolean isDefault;
+	private String rule;

```

Comment:
Where is `setRule()` and `setName()`? How can we edit this information without a set method?
Keep in mind that `setName()` should only update the name if `! isDefault`, and the `rule` as well. But the rule might be a bit more tricky to verify. Would need to check the number values and that it is the only thing that changed from the new to the old rule.

Reply:
Well, that verification shouldn't be made by the controller? Don't think we should make that on the setName() or setRule(). In my opinion the controller has to verify if that rule is default and disable the fields that we don't want to be editable

Reply:
I don't think it should come from the controller. Because this is a rule (no pun intended xD) from this class and it should make sure that it holds valid data. Maybe even, we could have a `isValidRule()` method that would validate a rule, or a `isValidNewRule()` that would check if it is a valid new rule and, in case this is a default rule, would compare against the current one and check if the new one only changed thresholds.

Reply:
I still suck at whole MVC thing, but from a logic standpoint I agree we should have setters here. However! No ifs. the method should be a simple setter, and we have to make sure that we only call it after doing the needed verifications in the controller.

Reply:
Ok, maybe I pushed the MVC model bit too hard and didn't give a good overview of it. There are somethings that should be in the controller, but it is the model that is the keeper of the data. From the first link:
> The model is ultimately gatekeeper for your data. You should be able to test your business logic without touching the UI at all.
An application might have different controllers, and if we would replicate the business logic in all of them, it would be a nightmare.
Also, the whole idea of having setters/getters is that we don't allow the user to just set values to whatever they want at will. Setters are there also for validation.

So if you want to keep this Class lean, I would suggest creating a business logic model that stands between this class and the controller, that is the holder of the validation model, to me called by the controller or whoever.

https://stackoverflow.com/a/4415941
https://softwareengineering.stackexchange.com/a/165446